library/psa_crypto_slot_management.c

```diff
@@ -236,7 +244,7 @@ static psa_status_t psa_internal_make_key_persistent( psa_key_handle_t handle,
 }
 
 static psa_status_t persistent_key_setup( psa_key_lifetime_t lifetime,
-                                          psa_key_id_t id,
+                                          psa_key_file_id_t id,
```

Comment:
Is changing `psa_key_id_t` to `psa_key_file_id_t` everywhere really the best option? 

If we keep this PR as-is and then do the following, we wouldn't require changing to `psa_key_file_id_t` everywhere. Would this be a workable improvement?
- Remove `typedef uint32_t psa_key_id_t;` from crypto_types.h
- Add `typedef psa_app_key_id_t psa_key_id_t` to crypto_platform.h for non-SPM platforms

Then we'd have the following situation:
- Using `psa_key_file_id_t` as `psa_key_id_t` on SPM platforms (what we sort of have in this PR now, but don't use it)
- Using `psa_app_key_id_t `as `psa_key_id_t` on non-SPM platforms

Maybe there is a reason to keep defining psa_key_id_t in crypto_types.h that I'm not seeing. In which case, we can keep that as-is and actually use that type definition instead of `psa_key_file_id_t` everywhere.

Reply:
Currently, `crypto_types.h` and `crypto.h` are used to generate the specification, so they have to use the standard definitions. I wanted to minimize the incursion of MbedCrypto-specific stuff there.

Once the specification and the implementation are fully split (which should take a couple of weeks), I think we should define `psa_key_file_id_t` in `crypto_types.h` and use it in `crypto.h`.